kie-server-parent/kie-server-api/pom.xml

```diff
@@ -24,11 +24,18 @@
       <artifactId>kie-api</artifactId>
       <!-- scope>provided</scope -->
     </dependency>
+
+    <!-- we need to find a way to remove the following 2 dependencies from this jar -->
     <dependency>
       <groupId>org.drools</groupId>
       <artifactId>drools-core</artifactId>
       <!-- scope>provided</scope -->
     </dependency>
+    <dependency>
+      <groupId>org.optaplanner</groupId>
+      <artifactId>optaplanner-core</artifactId>
+      <!-- scope>provided</scope -->
```

Comment:
very good point @sutaakar and looks like this was the reason build failed on jenkins. My wrong thinking was that optional applies only when used as dependency and thus would make it optional as transitive but anyway since it was compiled with reference (in import section) that would fail as well on runtime. So we have to go back to non optional dependency for optaplanner